getting-started/quickstart-guides/angular.md

```diff
@@ -0,0 +1,854 @@
+
+**Fetching uploaded file**
+
+Fetch uploaded image and show in UI using `ik-image` with the `filePath` returned in the upload response.
+
+```jsx
+<ik-image path="/test_RiCBw0ouh.png" ></ik-image>
+```
+
+The app should display your uploaded image correctly!
+
+## **Advanced file upload**
+
+A more detailed example of how to use the file upload component (and an explanation of each advanced feature) is presented below:
+
+{% code title="src/app/app.component.ts" %}
+```js
+import { Component } from '@angular/core';
+import { Transformation } from 'imagekit-javascript/dist/src/interfaces/Transformation';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+    validateFileFunction(res: any) {
+    console.log('validating')
+    if(res.size < 1000000){ // Less than 1mb
+      return true;
+    }
+    return false;
+  }
+
+  onUploadStartFunction(res: any) {
+    console.log('onUploadStart')
+  }
+
+  onUploadProgressFunction(res: any) {
+    console.log('progressing')
+  }
+
+  handleUploadSuccess(res: any) {
+    console.log('File upload success with response: ', res);
+    console.log(res.$ResponseMetadata.statusCode); // 200
+    console.log(res.$ResponseMetadata.headers); // headers
+    this.uploadedImageSource = res.url;
+  }
+
+  handleUploadError(err: any) {
+    console.log('There was an error in upload: ', err);
+    this.uploadErrorMessage = 'File upload failed.';
+  }
+}
+```
+{% endcode %}
+
+{% code title="src/app/app.component.html" %}
+{% endcode %}
+
+### **Custom Upload Button**
+We have created a `buttonRef` attribute for `ik-upload`. Using this, the file selector component inside ik-upload will be invisible and you can creaete your own `button`, with your own stylings, and reference it using `#<buttonRefValue>`. 
+
+Here's an example:
+
+```js
+<div>
+    <!-- This will be invisible -->
+    <ik-upload
+      [buttonRef]="myBtn"
+      >
+    </ik-upload>
+
+    <!-- Your own button! -->
+    <button #myBtn type="Button" class="myOwnClass" style="color:blue">
+      Upload
+    </button>
+</div>
+```
+
+### **Upload start**
+The `onUploadStart` prop is called when the file upload starts. This can be used for common use cases like showing a spinner, progress bar, etc.
+
+### **Show progress bar**
+The `onUploadProgress` prop can be passed like above, which will have a [ProgressEvent](https://developer.mozilla.org/en-US/docs/Web/API/ProgressEvent). This can be used to show the percentage of upload progress to the end user.
+
+### **Validate file before upload**
+Arbitrary validation (file type, file size, file name) etc can be added using the `validateFile` prop. An example has been added above that shows how to prevent upload if the file size is bigger than 2 MB.
```

Comment:
The code above is checking less than 1 MB and not 2MB.

Reply:
Resolved